src/nettools/ping.go

```diff
@@ -2,20 +2,108 @@ package nettools
 
 import (
 	"context"
-	"os/exec"
-	"runtime"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+
+	"golang.org/x/net/icmp"
+	"golang.org/x/net/ipv6"
+
+	"github.com/zerops-io/zcli/src/i18n"
 )
 
-func HasIPv6PingCommand() bool {
-	_, err := exec.LookPath("ping6")
-	return err == nil || runtime.GOOS == "windows"
-}
+const ProtocolIPv6ICMP = 58
 
 func Ping(ctx context.Context, address string) error {
-	pingCommand := exec.CommandContext(ctx, "ping6", "-c", "1", address)
-	if runtime.GOOS == "windows" {
-		pingCommand = exec.CommandContext(ctx, "ping", "/n", "1", address)
+	c, err := icmp.ListenPacket("ip6:ipv6-icmp", "::")
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	dst, err := net.ResolveIPAddr("ip6", address)
+	if err != nil {
+		return err
+	}
+
+	m := icmp.Message{
+		Type: ipv6.ICMPTypeEchoRequest,
+		Code: 0,
+		Body: &icmp.Echo{
+			ID:   os.Getpid() & 0xffff,
+			Seq:  1,
+			Data: []byte(""),
+		},
+	}
+
+	b, err := m.Marshal(nil)
+	if err != nil {
+		return err
+	}
+
+	n, err := c.WriteTo(b, dst)
+	if err != nil {
+		return err
+	} else if n != len(b) {
```

Comment:
do we need the `else if` ? it looks like the `if` is sufficient

Reply:
Yes, we check if `WriteTo` returned error and if not, that count of bytes written is equal to length of buffer.